jazz-build-module/scm-module.groovy

```diff
@@ -0,0 +1,179 @@
+                returnStdout: true
+            ).trim()
+
+            def jsonSlurper = new JsonSlurper()
+            def object = jsonSlurper.parseText(gitlab_repo_output)
+            
+            if(object == null || object.equals("") || object.id == null || object.id.equals("")){
+                error "project creation in gitlabs failed"
+            }
+             
+            cas_proj_id = object.id
+
+            cas_repo_id = getCasRepoId(gitlab_private_token, repo_loc)
+
+            transferProject(cas_repo_id, cas_proj_id)
+        }
+        else{
+            sh "curl -X POST -k -v -u \"${scm_config.BITBUCKET.BITBUCKET_USERNAME}:${scm_config.BITBUCKET.BITBUCKET_PASSWORD}\" -H \"Content-Type: application/json\" " + scm_user_services_api_endpoint + " -d \'{\"name\":\""+ repo_name +"\", \"scmId\": \"git\", \"forkable\": \"true\"}\'"
+        }
+    }catch (ex) {
+        if(!((ex.getMessage()).indexOf("groovy.json.internal.LazyMap") > -1)) {
+            echo "createUserInSCM Failed"
+            error "createProjectInSCM Failed. "+ex.getMessage()
+        }
+    }
+}
+
+def getGitlabUserId(gitlab_username){
+    try{
+        def output = sh (
+            script: "curl --header \"Private-Token: ${scm_config.GITLAB.gitlab_private_token}\" -X GET \"http://${scm_config.REPOSITORY.REPO_BASE}/api/v3/users?username=$gitlab_username\"",
+            returnStdout: true
+        ).trim()
+  
+        def jsonSlurper = new groovy.json.JsonSlurper()
+        def userObject = jsonSlurper.parseText(output)
+  
+        if(userObject == null || userObject.equals("") || userObject[0] == null || userObject[0].equals("") 
+            || userObject[0].id == null || userObject[0].id.equals("")){
+                error "get user data in gitlab failed"
+        }
+        
+        return userObject[0].id
+    }
+    catch (ex) {
+            if(!((ex.getMessage()).indexOf("groovy.json.internal.LazyMap") > -1)) {
+                echo "getGitlabUserId Failed"
+                error "getGitlabUserId Failed. "+ex.getMessage()
+            }
+        }
+}
+
+def getCasRepoId(repo_loc){
+    try{
+        def output = sh (
+            script: "curl --header \"Private-Token: ${scm_config.GITLAB.gitlab_private_token}\" -X GET \"http://${scm_config.REPOSITORY.REPO_BASE}/api/v3/groups?search=$repo_loc\"",
+            returnStdout: true
+        ).trim()
+
+        def jsonSlurper = new JsonSlurper()
+        def groupObject = jsonSlurper.parseText(output)
+        
+        if(groupObject == null || groupObject.equals("") || groupObject[0] == null || groupObject[0].equals("") 
+            || groupObject[0].id == null || groupObject[0].id.equals("")){
+            error "get cas group data failed"
+        }
+        
+        return groupObject[0].id
+    }catch (ex) {
+        if(!((ex.getMessage()).indexOf("groovy.json.internal.LazyMap") > -1)) {
```

Comment:
Refer comments above (line numbers - 95 $ 98)

Reply:
Fixed it by having a generic catch as none of the exceptions are actionable.